plugins/modules/public_cloud_instance.py

```diff
@@ -94,11 +109,38 @@ def run_module():
     service_name = module.params['service_name']
     flavor_id = module.params['flavor_id']
     image_id = module.params['image_id']
-    service_name = module.params['service_name']
     ssh_key_id = module.params['ssh_key_id']
     region = module.params['region']
     networks = module.params['networks']
     monthly_billing = module.params['monthly_billing']
+    force_reinstall = module.params['force_reinstall']
+
+
+    try:
+        instances_list = client.get('/cloud/project/%s/instance' % (service_name),
+                            region=region
+                            )
+    except APIError as api_error:
+        module.fail_json(msg="Failed to call OVH API: {0}".format(api_error))
+
+    for i in instances_list:
+
+        if i['name'] == name:
+            instance_id = i['id']
+            instance_details = client.get('/cloud/project/%s/instance/%s' % (service_name,instance_id ))
+
+            if force_reinstall:
+                try:
+                    reinstall_result = client.post(
+                                        '/cloud/project/%s/instance/%s/reinstall' % (service_name,instance_id),
+                                         imageId=image_id
+                                         )
+                    module.exit_json(changed=True, **reinstall_result)
+
+                except APIError as api_error:
+                    module.fail_json(msg="Failed to call OVH API: {0}".format(api_error))
+
+            module.exit_json(changed=False,**instance_details)
```

Comment:
Maybe we can exit with a message ? Like `module.exit_json(changed=False,msg="Instance {} is already installed".format(name)` ?